test/Compatibility/CompatibilityTests/CompatibilityTests.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "AssemblyInfo.h"
+
+#include <FrameworkUdk/Containment.h>
+#include <winrt/Microsoft.Windows.ApplicationModel.WindowsAppRuntime.h>
+
+namespace TB = ::Test::Bootstrap;
+namespace TP = ::Test::Packages;
+
+namespace WAR = winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime;
+
+namespace Test::CompatibilityTests
+{
+    class CompatibilityTests
+    {
+    public:
+        BEGIN_TEST_CLASS(CompatibilityTests)
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA") // MTA is required for ::Test::Bootstrap::SetupPackages()
+            TEST_CLASS_PROPERTY(L"RunFixtureAs:Class", L"RestrictedUser")
+            TEST_CLASS_PROPERTY(L"IsolationLevel", L"Method") // each test sets its own CompatibilityOptions
+        END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassSetup)
+        {
+            ::Test::Bootstrap::SetupPackages();
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassCleanup)
+        {
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppRuntimeFramework());
+
+            // The test method setup and execution is on a different thread than the class setup.
+            // Initialize the framework for the test thread.
+            ::Test::Bootstrap::SetupBootstrap();
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            VERIFY_IS_TRUE(TP::IsPackageRegistered_WindowsAppRuntimeFramework());
+            ::Test::Bootstrap::CleanupBootstrap();
+            return true;
+        }
+
```

Comment:
would be good to test error conditions to ensure well-defined behavior (e.g., calling Apply twice, incompatible modes, etc)